app/channel.go

```diff
@@ -2614,7 +2614,19 @@ func (a *App) createInitialSidebarCategories(user *model.User, team *model.Team)
 }
 
 func (a *App) GetSidebarCategories(userId, teamId string) (*model.OrderedSidebarCategories, *model.AppError) {
-	return a.Srv().Store.Channel().GetSidebarCategories(userId, teamId)
+	categories, err := a.Srv().Store.Channel().GetSidebarCategories(userId, teamId)
+
+	if len(categories.Categories) == 0 && err == nil {
+		// A user must always have categories, so migration must not have happened yet, and we should run it ourselves
+		nErr := a.createInitialSidebarCategories(userId, teamId)
+		if nErr != nil {
+			return nil, nErr
+		}
+
+		categories, err = a.Srv().Store.Channel().GetSidebarCategories(userId, teamId)
```

Comment:
Is replica lag going to be a problem here?

Reply:
Ah I see you commented on this. Can we do some kind of wait-loop here? I think we've done that elsewhere with joining users to teams in the past? I.e. check if the result is still zero, and if it is, wait a bit and try again a couple of times, only then giving up and returning no categories.

Reply:
Yep, I'll take a look at that. If you happen to have any references on this, could you DM me a link?

Reply:
Never mind. Found one: https://github.com/mattermost/mattermost-server/blob/master/app/channel.go#L393

Reply:
Ahh here we are :)

Reply:
So this maybe a stupid question, but it seems to me that we are doing read-after-write within a single request. Why can't 

```
nErr := a.createInitialSidebarCategories(userId, teamId)
		if nErr != nil {
			return nil, nErr
		}
```

return the categories itself?

It seems like we are creating a resource, but not returning that in that response itself, but rather again reading from the DB using the same identifier.